great_tables/_tbl_data.py

```diff
@@ -249,17 +249,20 @@ def _(data: PdDataFrame, group_key: str) -> Dict[Any, List[int]]:
 
 @group_splits.register
 def _(data: PlDataFrame, group_key: str) -> Dict[Any, List[int]]:
-    # TODO: should ensure row count name isn't already in data
     import polars as pl
 
+    _row_count_name = "__row_count__"
+    if _row_count_name in data.columns:
+        raise ValueError(f"{_row_count_name} is already in data.columns")
+
     # with_row_index supercedes with_row_count
-    meth_row_number = getattr(data, "with_row_index", None)
-    if not meth_row_number:
-        meth_row_number = data.with_row_count
+    meth_row_number = getattr(data, "with_row_index", "with_row_count")
```

Comment:
The code will return the string "with_row_count", when it should be returning a .with_row_count method.